### PR TITLE
[docs] New contributing doc on How to Open a Pull Request

### PR DESCRIPTION
**Pull request by marcysutton**
## Description

There have been some comments recently (including in the core maintainers meeting, I believe) that a page on How to Open a Pull Request would be helpful, so I created one! I moved some of the content from the [Managing PRs](https://gatsbyjs.org/contributing/managing-pull-requests/) doc and added a link back to it, as well as updating a few surrounding docs. Feedback welcome on the content -- I sorta just pushed what I had so we could iterate on it.

There's also a section on blog post submissions @caitlinthefirst, so it would be good to chat about our documented workflow/policy for that and potentially link to it here. But that could also come in a follow-up PR.

## Related Issues

N/A

### CODE REVIEW

**Comment by universse** [comment]
Hi I think this is a good resource as well http://makeapullrequest.com/. It's an egghead video course, also helpful for people new to git in general.

**Comment by muescha** [comment]
Since there are many commits everyday to the repo.: would it be good also to add an section how to have my forked repo up to date with the source repo?

**Comment by marcysutton** [comment]
Good idea, @muescha, will add. Thanks!

**Comment by muescha** [comment]
Also a question i come over: 
since many upgraded as contributors of Gatsby:

1) Should I still fork the repo (and have the sync problem) and make PR from my repo
2) should I clone the repo and make PR from.my feature branch

What is the official/best way to go?

File: docs/contributing/docs-contributions.md [hunk 1]
```diff
@@ -91,17 +91,17 @@ If you wrote a new document that was [previously a stub](/contributing/how-to-wr
 
 ## Docs site setup instructions
 
-After going through the development setup instructions above, there are a few additional things that are helpful to know when setting up the [Gatsby.js docs site](/docs/). which mostly lives in the [www](https://github.com/gatsbyjs/gatsby/tree/master/www) directory.
+After going through the [development setup instructions](/contributing/setting-up-your-local-dev-environment/), there are a few additional things that are helpful to know when setting up the [Gatsby.js docs site](/docs/). which mostly lives in the [www](https://github.com/gatsbyjs/gatsby/tree/master/www) and [docs](https://github.com/gatsbyjs/gatsby/tree/master/docs) directories. There are also some [examples](https://github.com/gatsbyjs/gatsby/tree/master/examples) in the repo that are referenced in docs.
```
Comment (gillkyle) [inline docs/contributing/docs-contributions.md]:
```suggestion
After going through the [development setup instructions](/contributing/setting-up-your-local-dev-environment/), there are a few additional things that are helpful to know when setting up the [Gatsby.js docs site](/docs/), which mostly lives in the [www](https://github.com/gatsbyjs/gatsby/tree/master/www) and [docs](https://github.com/gatsbyjs/gatsby/tree/master/docs) directories. There are also some [examples](https://github.com/gatsbyjs/gatsby/tree/master/examples) in the repo that are referenced in docs.
```

**Comment by marcysutton** [comment]
This one is ready for another review! cc @gillkyle

File: docs/contributing/how-to-open-a-pull-request.md [hunk 2]
```diff
@@ -80,6 +81,40 @@ There are specific pages about contributing to various parts of the Gatsby ecosy
...
+  ```
+  - If there are any [merge conflicts](https://help.github.com/en/articles/resolving-a-merge-conflict-on-github), you'll want to address those to get a clean merge.
+- Once your branch is in good working order, push the changes to your fork:
+  ```bash
+  git push origin head
+  ```
+
+For more information on working with upstream repos, [visit the GitHub docs](https://help.github.com/en/articles/configuring-a-remote-for-a-fork).
+
+_**Note:** as a member of the Gatsby repo, you can also clone it directly and push to feature branches._
```
Comment (muescha) [inline docs/contributing/how-to-open-a-pull-request.md]: Maybe add some hint what I have to change with the commands if Ihave fetch and merge into
A feature branch?

**Pull request merged**